### DISCUSSION TITLE
How do I config the Renovate uses the ssh key to access the gitlab instance in shell script

### DISCUSSION THREAD

**Discussion by xiujuanx** [renovatebot/renovate#25128] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.87.7

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

Placeholder value, please select the correct response from the dropdown

### Wanted end result.

My repo is hosted on enterprise gitlab and I am using my personal access token to access gitlab instance. The config is like the below:
```
    endpoint: 'https://gitlab.eng.vmware.com/api/v4/',
    platform: 'gitlab',
    token: process.env.RENOVATE_TOKEN,
```
But I don't want to use my personal account to raise MRs, I would like to use a bot. I search the gitlab doc, there have two choices for us, one is the [deploy key](https://docs.gitlab.com/ee/user/project/deploy_keys/#create-a-project-deploy-key), another one is the [service account](https://docs.gitlab.com/ee/user/profile/service_accounts.html#self-managed-gitlab). Enable a deploy key is more simple for us, however, if we use the deploy key, then we need config the Renovate to consume it. However, I can't find any document about how to use a ssh key for Renovate in shell script. So I am asking this question. Could anyone help us on this issue? Thanks!

### What you tried so far.

The personal access token works, but I prefer to using the ssh key for Renovate.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment, marked as answer]
ssh isn't supported

**Reply by xiujuanx** [reply]
I see, thanks!
